devenv/lib/volta.py

```diff
@@ -23,49 +23,66 @@ class UnexpectedPlatformError(Exception):
     pass
 
 
-def install_volta(unpack_into: str) -> None:
+def determine_platform(unpack_into: str) -> str | None:
     system = platform.system()
     if system == "Linux":
         if platform.machine() == "x86_64":
-            name = f"volta-{_version}-linux.tar.gz"
+            return f"volta-{_version}-linux.tar.gz"
         else:
             proc.run(("brew", "install", "volta"), exit=True)
             proc.run(
                 ("sh", "-c", f"ln -sfn {homebrew_bin}/volta* {unpack_into}/"),
                 exit=True,
             )
-            return
+            return None
     elif system == "Darwin":
         suffix = "-aarch64" if platform.machine() == "arm64" else ""
-        name = f"volta-{_version}-macos{suffix}.tar.gz"
+        return f"volta-{_version}-macos{suffix}.tar.gz"
     else:
         raise UnexpectedPlatformError(f"Unexpected OS: {platform.platform()}")
 
-    url = (
+
+def build_url(name: str) -> str:
```

Comment:
sorry i missed this one - i don't think we should have a function merely formatting a string and used once

as for the tests, i would assert that `archive.download` was called with the expected url

Reply:
Fair point